File: client/src/types/message.ts
import { Base } from "./base";

export type Message = {
  content: string;
  userId: number;
  conversationId: number;
} & Base;

export type MessageRequest = {
  content: string;
  conversationId: number;
};

export type MessageResponse = {
  message: Message;
  messageBot: Message;
};

File: server/internal/entities/message.go
package entities

import "time"

type Message struct {
	Id             uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	Content        string    `json:"content" `
	UserId         uint      `json:"userId" gorm:"default:null"`
	ConversationId uint      `json:"conversationId"`
	CreatedAt      time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt      time.Time `json:"updated_at" gorm:"autoUpdateTime"`
}
